src/main/kotlin/org/rust/lang/core/resolve/ref/RsMarcoItemReferenceContributor.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.resolve.ref
+
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.*
+import com.intellij.util.ProcessingContext
+import org.rust.lang.core.psi.RsElementTypes.IDENTIFIER
+
+class RsMarcoItemReferenceContributor : PsiReferenceContributor() {
+    override fun registerReferenceProviders(registrar: PsiReferenceRegistrar) {
+        val pattern = PlatformPatterns.psiElement(IDENTIFIER)//never called
```

Comment:
I want to use my current process alg. To write an completion contributor. Else I need to write some parts more than once.